src/relayclient/RelayProvider.ts

```diff
@@ -25,39 +26,36 @@ export type JsonRpcCallback = (error: Error | null, result?: JsonRpcResponse) =>
 interface ISendAsync {
   sendAsync?: any
 }
-
-export class RelayProvider implements HttpProvider {
+// TODO: stop faking the HttpProvider implementation -  it won't work for any other 'origProvider' type
+export class RelayProvider implements HttpProvider, ISendProvider {
   protected readonly origProvider: HttpProvider & ISendAsync
   private readonly origProviderSend: any
   protected config!: GSNConfig
 
   readonly relayClient: RelayClient
   logger!: LoggerInterface
 
+  static newProvider (input: GSNUnresolvedConstructorInput): RelayProvider {
```

Comment:
the minimal construction becomes:
```
RelayProvider.newProvider({ provider:window.ethereum, partialConfig:{paymasterAddress: '0x1234'}})
```

Do you think its nice?

Reply:
I mean, from the user's perspective, it should be :
```
RelayProvider.newProvider({ provider:window.ethereum, paymasterAddress: '0x1234'})
```
the 2nd nesting of "partialConfig" is completely useless (I don't care about overrideDependencies, since that's something for testing only)

Reply:
Well, I strongly suggest creating/reading the configuration into a variable(s) before passing it to the method. 
I write with this mentality, and yes, different code styles will vary in how well it looks.

```
const provider = window.ethereum
const partialConfig: Partial<GSNConfig> = { payamsterAddress: '0x123123' }
const relayProvider: RelayProvider = RelayProvider.newProvider({ provider, partialConfig })
```

I feel this code is way more explicit, readable and easier to work with.